src/isoStringScalar/isoStringScalar.ts

```diff
@@ -0,0 +1,86 @@
+import { GraphQLScalarType, Kind } from 'graphql';
+import { isValid, parseISO, toDate } from 'date-fns';
+import {
+  InternalServerError,
+  UserInputError,
+} from '../errorHandler/errorHandler';
+
+export const isoStringScalar = new GraphQLScalarType({
+  name: 'ISOString',
+  description: 'ISOString custom scalar type',
+
+  /**
+   * Converts server-side TS Date representation to a JSON-compatible, ISO-8601-compliant
+   * UTC Datetime String for Apollo Server to include in an operation response.
+   *
+   * @param value - TS Date Object as generated by Data Store Client & validated by mysqlDateConvert.
+   * @returns ISO-8601-compliant, UTC-based Datetime String.
+   */
+  serialize(value: Date | null): string | null {
+    if (value === null) {
+      return null;
+    }
+
+    if (value instanceof Date && isNaN(value.valueOf())) {
+      throw new InternalServerError(
+        'Invalid Data Store Response: invalid Date object'
+      );
+    }
+
+    if (value instanceof Date) {
+      // toISOString() is always zero UTC offset per
+      // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
+      const resp: string = value.toISOString();
+      return resp;
+    }
+
+    throw new InternalServerError(
+      'GraphQL ISOString Scalar serializer expected a `Date` object or null'
+    );
+  },
+
+  /**
+   * Converts the scalar's JSON value to its back-end representation
+   * before it's added to a resolver's args.
+   *
+   * @param value - ISO-8601-compliant Datetime String.
+   * @returns TS Data Object for interacting with Data Store via Client.
+   */
+  parseValue(value: string | null): Date | null {
+    if (value === null || value === '') {
+      return null;
+    }
+
+    if (typeof value === 'string') {
```

Comment:
Huh, github gobbled one of my comments.  Same general optional style suggestion here.  Invert your assertions so that happy path isn't in a branch, and the end of the function is returning the intended result.

Reply:
haha just did this looking at your other comments. thanks for the help!